src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -238,18 +239,35 @@ export default class AmalgamatingBusinesses extends Mixins(AmalgamationMixin, Co
 
   /** TextField rules for "Add an Amalgamating Foreign Business" Panel. */
   get foreignBusinessLegalNameRules (): Array<(v) => boolean | string> {
-    return [ v => !!v || 'Full legal name is required' ]
+    return [
+      v => !!v || 'Full legal name is required',
+      v => (!v || v.length >= 3) || 'Must be at least 3 characters',
+      v => (!v || v.length <= 150) || 'Cannot exceed 40 characters'
+    ]
   }
 
   get foreignBusinessCorpNumberRules (): Array<(v) => boolean | string> {
-    return [ v => !!v || 'Corporate number is required' ]
+    return [
+      v => (!this.isMrasJurisdiction || (!!v && /^[0-9a-zA-Z-]+$/.test(v))) ||
+        'Corporate number is required',
+      v => (!v || v.length >= 3) || 'Must be at least 3 characters',
+      v => (!v || v.length <= 40) || 'Cannot exceed 40 characters'
+    ]
   }
 
   /** Called when Jurisdiction menu item is changed. */
   onJurisdictionChange (jurisdiction: any): void {
     this.jurisdiction = jurisdiction
     this.isCan = jurisdiction.group === 0
-    this.jurisdictionErrorMessage = this.jurisdiction ? '' : 'Required.'
+    this.jurisdictionErrorMessage = this.jurisdiction ? '' : 'Home jurisdiction is required'
+    this.isMrasJurisdiction = ['AB', 'MB', 'NS', 'ON', 'QC', 'SK'].includes(
```

Comment:
It would be better to declare this externally/centrally. Could the following be used (perhaps in conjunction with the `CanJurisdictions` array)?

https://github.com/bcgov/bcrs-shared-components/blob/2cdc2aeea28ac1f91df7cf21c8573f4ae5e94b6f/src/components/jurisdiction/list-data/canada-jurisdictions.ts#L77

Reply:
Yup, updated!